components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2901,25 +2901,18 @@ where
                 observe_id,
                 region_id,
                 enabled,
-            } => {
-                assert!(
-                    !self
-                        .delegate
-                        .observe_cmd
-                        .as_ref()
-                        .map_or(false, |o| o.enabled.load(Ordering::SeqCst)),
-                    "{} observer already exists {:?} {:?}",
-                    self.delegate.tag,
-                    self.delegate.observe_cmd,
-                    observe_id
-                );
-                (observe_id, region_id, Some(enabled))
-            }
+            } => (observe_id, region_id, Some(enabled)),
             ChangeCmd::Snapshot {
                 observe_id,
                 region_id,
             } => (observe_id, region_id, None),
         };
+        if let Some(observe_cmd) = self.delegate.observe_cmd.as_mut() {
+            if observe_cmd.id > observe_id {
+                notify_stale_req(self.delegate.term, cb);
```

Comment:
Stale error will be translated into region not found error is it ok?
https://github.com/tikv/tikv/blob/cdca278ba7c2f631d5b3b5db08f1335dc7a89108/components/cdc/src/delegate.rs#L292-L311

Reply:
It is ok for now, the client will treat those errors in a same way.